app/src/cache.js

```diff
@@ -117,17 +118,22 @@ export default class FileCache {
         }
     }
 
+    // Return the path where the given Archive file is downloaded to.
+    // (HASH.zip or HASH.tar.gz in the cache dir.)
     file_path(hash, type) {
         return path.join(this.cache_dir, `${hash.toString(36)}.${type}`)
     }
 
     // Get a file out of the cache, or download it
+    // Note that this may return a promise! The caller must await it for a valid entry.
```

Comment:
Async functions *always* return a promise, and must always be `await`ed/`.then`ed etc. The magic of the cache is that it may return a promise that resolves immediately into the cache, or it returns a promise the resolves into a promise that will resolve when the file has finished downloading. The caller doesn't need to care however, as all levels of promises get flattened by `await`.

Reply:
You can tell I never got entirely up to speed on promises. `:)` Feel free to edit.

Reply:
To me the function signature is indication enough (and there are a bunch of other async functions too) so I wouldn't think any comment is necessary. But if you wanted something here I guess it could say "This may immediately return the file entry, or it may return a promise that waits for it to be downloaded, but you don't need to worry about"